```
[ERROR]   SabreV4ResourceAMQPMessageTest.shouldReceiveMessageFromEventResourceAcceptExchange » Runtime
[ERROR]   SabreV4ResourceAMQPMessageTest.shouldReceiveMessageFromEventResourceDeclineExchange » Runtime
```

I seems this change affect resource validation flow.

If principal is a resource we can likely trigger regular flow.

```
[ERROR]   SabreV4CalDavTest.shouldBeAbleToAcceptParticipationOfResourceByUsingToken » Runtime
```

This one is less clear to me...

--- a/lib/CalDAV/Schedule/Plugin.php
+++ b/lib/CalDAV/Schedule/Plugin.php
@@ -194,6 +194,7 @@ class Plugin extends \Sabre\CalDAV\Schedule\Plugin {
      * 1. If there are no significant changes ($changes is empty) → skip
      * 2. If the recipient is an attendee and the change concerns another attendee's
      *    participation status → skip (only organizer needs this info)
+     * 3. If the recipient is a resource → never skip (resources need all notifications)
      *
      * @param ITip\Message $message The IMIP message to evaluate
      * @param VCalendar|null $oldObject The original calendar object
@@ -214,6 +215,12 @@ class Plugin extends \Sabre\CalDAV\Schedule\Plugin {
         // Ensure oldObject has VEVENT
         if (!isset($oldObject->VEVENT) || !isset($newObject->VEVENT)) {
             return false;
+        }
+
+        // Check if recipient is a resource - resources need all notifications
+        $recipientPrincipalUri = \ESN\Utils\Utils::getPrincipalByUri($message->recipient, $this->server);
+        if ($recipientPrincipalUri && \ESN\Utils\Utils::isResourceFromPrincipal($recipientPrincipalUri)) {
+            return false; // Never skip for resources
         }
 
         // Get the organizer email

--- a/lib/CalDAV/Schedule/Plugin.php
+++ b/lib/CalDAV/Schedule/Plugin.php
@@ -206,6 +206,16 @@ class Plugin extends \Sabre\CalDAV\Schedule\Plugin {
             return false;
         }
 
+        // Parse oldObject if it's a string (raw iCalendar data)
+        if (is_string($oldObject)) {
+            $oldObject = \Sabre\VObject\Reader::read($oldObject);
+        }
+
+        // Ensure oldObject has VEVENT
+        if (!isset($oldObject->VEVENT) || !isset($newObject->VEVENT)) {
+            return false;
+        }
+
         // Get the organizer email
         $organizerEmail = null;
         if (isset($newObject->VEVENT->ORGANIZER)) {
@@ -243,7 +253,7 @@ class Plugin extends \Sabre\CalDAV\Schedule\Plugin {
      * @param \Sabre\VObject\Component $newEvent
      * @return bool True if there are significant changes
      */
-    private function hasSignificantPropertyChanges($oldEvent, $newEvent): bool {
+    private function hasSignificantPropertyChanges(\Sabre\VObject\Component $oldEvent, \Sabre\VObject\Component $newEvent): bool {
         $significantProperties = [
             'DTSTART', 'DTEND', 'DURATION', 'SUMMARY', 'DESCRIPTION',
             'LOCATION', 'RRULE', 'EXDATE', 'RDATE', 'SEQUENCE'

